server/routes/imageUploaderAPI.js

```diff
@@ -1,37 +1,50 @@
-const express = require('express');
+const express = require("express");
 const router = express.Router();
 
-const multer = require('multer');
-const multerS3 = require('multer-s3');
-const AWS = require('aws-sdk');
+const multer = require("multer");
+const multerS3 = require("multer-s3");
+const AWS = require("aws-sdk");
+
+const auth = require("../middleware/auth");
+const db = require("../db/models");
+const mongoose = require("mongoose");
 
 // Set the region
 AWS.config.update({ region: process.env.AWS_REGION });
 
+const BUCKET = process.env.AWS_BUCKET_NAME;
+
 // Create S3 service object
 const s3 = new AWS.S3();
-const FN = Date.now().toString() + '.png'
+
 // AWS Params put into storage using Multer S3
 const upload = multer({
   storage: multerS3({
     s3: s3,
-    bucket: process.env.AWS_BUCKET_NAME,
+    bucket: BUCKET,
     key: function (req, file, cb) {
+      const FN = Date.now().toString() + ".png";
       cb(null, FN);
     },
   }),
-}).single('file');
+}).single("file");
 
 // Sends Storage Object to S3 Bucket
-router.post('/api/image-upload', (req, res) => {
+router.post("/api/image-upload", auth, (req, res) => {
   upload(req, res, (err) => {
     if (err) {
-      res.status(500).send('There was a server error with the upload');
+      res.status(500).send("There was a server error with the upload");
     } else {
       if (req.file == undefined) {
-        res.status(400).send('There was a error with your upload paramaters');
+        res.status(400).send("There was a error with your upload paramaters");
       } else {
-        res.status(200).send({msg:'Your image was uploaded sucessfully', fileName : req.file.key});
+        const uploadRes = res;
```

Comment:
I don't think it's necessary to rename the response here, the reason your original solution wasn't working was because adding `res` as an argument in the .then() reassigned the value

Reply:
That makes sense!